src/NuGet.Core/NuGet.Packaging/Licenses/LicenseExpressionTokenizer.cs

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace NuGet.Packaging.Licenses
+{
+    internal class LicenseExpressionTokenizer
+    {
+        private readonly string _value;
+
+        /// <summary>
+        /// A tokenizer for a license expression.
+        /// This implementation assumes that the input has been sanitized and that there are no invalid characters.
+        /// </summary>
+        /// <param name="value"></param>
+        internal LicenseExpressionTokenizer(string value)
+        {
+            if (string.IsNullOrWhiteSpace(value))
+            {
+                throw new ArgumentException(nameof(value));
+            }
+            _value = value.Trim();
+        }
+
+        /// <summary>
+        /// The valid characters for a license identifier are a-zA-Z0-9.-+
+        /// The valid characters for a license expression are the above whitespace and ().
+        /// </summary>
+        /// <param name="value"></param>
+        /// <returns></returns>
```

Comment:
fill out `<param` and `<returns>` tags or remove them if they do not provide useful information.